app/controllers/tickets_controller.rb

```diff
@@ -19,6 +19,8 @@ class TicketsController < ApplicationController
   include TicketsStrongParams
   include ActionView::Helpers::SanitizeHelper # dependency of HtmlTextHelper
 
+  MAIL_KEY = Digest::MD5.hexdigest(Rails.application.secrets.secret_key_base).freeze
```

Comment:
Maybe you should use something newer, more secure, because MD5 [has been broken](http://cryptocrats.com/crypto/md5-the-hash-algorithm-is-now-broken/) and you're basically leaking the `secret_key_base`.

Reply:
I meant it's quite easy to brute-force, especially if you know the length of the input.

Reply:
7.922816251e28 possibilities on a web server is not really easy, but you're right, 6.277101735e57 possibilities are even harder to cover. :smile: